lib/dialects/abstract/connection-manager.js

```diff
@@ -184,11 +184,28 @@ ConnectionManager.prototype.getConnection = function(options) {
   var self = this;
   options = options || {};
 
-  return new Promise(function (resolve, reject) {
-    self.pool.acquire(function(err, connection) {
-      if (err) return reject(err);
-      resolve(connection);
-    }, options.priority, options.type, options.useMaster);
+  return Promise.resolve().then(function () {
+    if (self.sequelize.options.databaseVersion === 0) {
```

Comment:
Just if you have several concurrent calls i assume? Each will request a new connection from the pool.
Blocking would not be too tricky, save the connect + databaseversion promise in a variable, if var is set use that to resolve.